src/p2p/net_node.inl

```diff
@@ -1036,15 +1036,16 @@ namespace nodetool
         return;
       }
       hsh_result = true;
-      if(!just_take_peerlist)
+
+      if(!m_payload_handler.process_payload_sync_data(rsp.payload_data, context, just_take_peerlist, true))
```

Comment:
What's the logic behind just_take_peerlist being equivalent to just_check_height?

Reply:
It's really "close the connection when done". It used to be to get the peer list, it is also done for checking height now.